File: src/main/java/com/example/polimarche_api/service/sensor_service/PositionService.java
package com.example.polimarche_api.service.sensor_service;

import com.example.polimarche_api.model.sensor.Current;
import com.example.polimarche_api.model.sensor.Position;
import com.example.polimarche_api.model.sensor.dto.PositionDTO;
import com.example.polimarche_api.model.sensor.dto.mapper_dto.PositionDTOMapper;
import com.example.polimarche_api.repository.sensor_repository.PositionRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class PositionService {

    private final PositionRepository positionRepository;

    private final PositionDTOMapper positionDTOMapper = new PositionDTOMapper();

    public PositionService(PositionRepository positionRepository) {
        this.positionRepository = positionRepository;
    }


    public List<PositionDTO> getPositionBySessionAndSetup(Integer sessionId, Integer setupId) {
        return positionRepository.findBySessioneIdAndSetupId(sessionId, setupId).
                stream().
                map(positionDTOMapper).
                collect(Collectors.toList());
    }
}

File: lib/service/sensor/position_service.dart
import '../../model/sensor/position_model.dart';
import '../../repos/sensor/position_repo.dart';

class PositionService {
  final PositionRepo _positionRepo = PositionRepo();

  Future<List<Position>> getPositionDataBySessionAndSetupId(
      int sessionId, int setupId) async {
    return await _positionRepo.getPositionDataBySessionAndSetupId(
        sessionId, setupId);
  }
}
